src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -336,12 +336,14 @@ public IEnumerable<PackageItemViewModel> GetCurrent()
 
                     if (listItem.PackageLevel == PackageLevel.TopLevel)
                     {
-                        listItem.UpdatePackageStatus(_installedPackages);
+                        Task.Run(() => listItem.UpdatePackageStatus(_installedPackages))
```

Comment:
This seems to be called in a loop for every package, so I'm wondering if we end up spamming the thread pool a bit too much here. 

It may show up negatively on slower machines.

The other thing that's interesting is that UpdatePackageStatus has a bunch of RunAsync which probably some of those tasks get completed on the current threadpool thread which is why takes longer, but it does signal to me that we can remove some of those calls. 

An idea would be to make UpdatePackageStatus async itself. 

The calling method here can probably be made async as well, since we seem to have `async -> sync -> async over sync with RunAsync` .

Reply:
Great idea! I'll work on implementing these suggestions. Thank you!